### PR TITLE
DOC: Resolve RT03 errors for selected methods

### PR DESCRIPTION
**Pull request by bergnerjonas**
Resolve all RT03 errors for the following cases:

scripts/validate_docstrings.py --format=actions --errors=RT03 pandas.DataFrame.expanding
scripts/validate_docstrings.py --format=actions --errors=RT03 pandas.DataFrame.filter
scripts/validate_docstrings.py --format=actions --errors=RT03 pandas.DataFrame.first_valid_index
scripts/validate_docstrings.py --format=actions --errors=RT03 pandas.DataFrame.last_valid_index
scripts/validate_docstrings.py --format=actions --errors=RT03 pandas.DataFrame.get
- xref DOC: fix RT03 errors in docstrings  #57416
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by bergnerjonas** [comment]
Any idea why the unit tests / Numpy Dev failed? I only changed things in the docstrings not in the code itself.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5378,11 +5383,12 @@ def filter(
         axis : {0 or 'index', 1 or 'columns', None}, default None
             The axis to filter on, expressed either as an index (int)
             or axis name (str). By default this is the info axis, 'columns' for
-            DataFrame. For `Series` this parameter is unused and defaults to `None`.
+            `DataFrame`. For `Series` this parameter is unused and defaults to `None`.
```
Comment (mroeschke) [inline pandas/core/generic.py:5386]:
```suggestion
            ``DataFrame``. For ``Series`` this parameter is unused and defaults to ``None``.
```

File: pandas/core/generic.py [hunk 2]
```diff
@@ -4183,15 +4183,19 @@ def get(self, key, default=None):
...
         Get item from object for given key (ex: DataFrame column).
 
-        Returns default value if not found.
+        Returns ``default`` value if not found.
 
         Parameters
         ----------
         key : object
+            Key for which item should be returned.
+        default : object
```
Comment (mroeschke) [inline pandas/core/generic.py:4192]:
```suggestion
        default : object, default None
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @bergnerjonas

**Pull request merged**